product_route_profile/models/product_template.py

```diff
@@ -32,3 +34,13 @@ def _compute_route_ids(self):
                 rec.route_ids = [(6, 0, self.route_profile_id.route_ids.ids)]
             else:
                 rec.route_ids = False
+
+    def _search_route_ids(self, operator, value):
+        profiles = self.env["route.profile"].search([("route_ids", operator, value)])
```

Comment:
You can avoid a big search doing this directly in the return:

`
return [
"|",
            ("force_route_profile_id.route_ids", operator, value),
            "&",
            ("force_route_profile_id", "=", False),
            ("route_profile_id.route_ids", operator, value),
]
`

Reply:
done, I also squashed the commits.